### PR TITLE
[stable-1] Various backports from community.postgres 

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
Backports of
https://github.com/ansible-collections/community.postgresql/pull/32
https://github.com/ansible-collections/community.postgresql/pull/46
https://github.com/ansible-collections/community.postgresql/pull/51
https://github.com/ansible-collections/community.postgresql/pull/43
https://github.com/ansible-collections/community.postgresql/pull/42
https://github.com/ansible-collections/community.postgresql/pull/36
https://github.com/ansible-collections/community.postgresql/pull/52

##### ISSUE TYPE
- Bugfix Pull Request
- Docs Pull Request

##### COMPONENT NAME
postgresql modules

### CODE REVIEW

**Comment by felixfontein** [comment]
This should also restart the AZP tests :)

File: plugins/modules/database/postgresql/postgresql_set.py [hunk 1]
```diff
@@ -232,34 +237,56 @@ def pretty_to_bytes(pretty_val):
     # The function returns a value in bytes
     # if the value contains 'B', 'kB', 'MB', 'GB', 'TB'.
     # Otherwise it returns the passed argument.
+    # It's sometimes possible to have an empty values
```
Comment (felixfontein) [inline plugins/modules/database/postgresql/postgresql_set.py:240]:
```suggestion

    # It's sometimes possible to have an empty values
```

File: plugins/modules/database/postgresql/postgresql_set.py [hunk 2]
```diff
@@ -310,9 +337,13 @@ def main():
 
     # Allow to pass values like 1mb instead of 1MB, etc:
```
Comment (felixfontein) [inline plugins/modules/database/postgresql/postgresql_set.py:338]:
```suggestion
```

File: changelogs/fragments/1-community-postgresql_backports.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+bugfixes:
+- postgresql_query - fix decimal handling (https://github.com/ansible-collections/community.postgresql/issues/45).
+- postgresql_query - fix datetime.timedelta type handling (https://github.com/ansible-collections/community.postgresql/issues/47).
+- postgresql_ping - fix crash caused by wrong PgSQL version parsing (https://github.com/ansible-collections/community.postgresql/issues/40).
+- postgresql_info - fix crash caused by wrong PgSQL version parsing (https://github.com/ansible-collections/community.postgresql/issues/40).
+- postgresql_set - return a message instead of traceback when a passed parameter has not been found (https://github.com/ansible-collections/community.postgresql/issues/41).
+- postgresql_set - fails in check_mode on non-numeric values containing `B` (https://github.com/ansible-collections/community.postgresql/issues/48).
```
Comment (felixfontein) [inline changelogs/fragments/1-community-postgresql_backports.yml:7]:
```suggestion
- postgresql_set - fails in check_mode on non-numeric values containing ``B`` (https://github.com/ansible-collections/community.postgresql/issues/48).
```

**Review by felixfontein — COMMENTED** [review]
I manually compared all code changes. I found two small differences, my suggestion would be to also apply them so the code is more similar (might make it easier to apply more backports in the future).

And then there's a detail in the changelog ;-)

**Comment by Andersson007** [comment]
> I manually compared all code changes. I found two small differences, my suggestion would be to also apply them so the code is more similar (might make it easier to apply more backports in the future).
> 
> And then there's a detail in the change

All committed, thanks very much for comparing!:)

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@Andersson007 thanks for backporting these fixes!

**Comment by Andersson007** [comment]
@felixfontein thanks for reviewing and merging!

**Pull request merged**